From aa792382bc23ad112cdc426c8ab9d926f6a4fc8b Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 20 Jun 2022 15:44:04 +0200
Subject: [PATCH] module-droid-*: use PA_MAJORMINOR as PA_MODULE_VERSION
 instead of PACKAGE_VERSION

* latest version is currently 14.2.97, but we need to build this for
  pulseaudio 15.0 in kirkstone and 16.0 in langdale

  use PA_MAJORMINOR which matches the version of libpulse we're building
  against:

  m4_define(pa_major, `pkg-config --modversion libpulse | cut -d. -f1 | cut -d- -f1`)
  m4_define(pa_minor, `pkg-config --modversion libpulse | cut -d. -f2 | cut -d- -f1`)
  ...
  AC_SUBST(PA_MAJOR, pa_major)
  AC_SUBST(PA_MAJORMINOR, pa_major.pa_minor)

  instead of pa_module_version which is from ./git-version:
  pulseaudio-modules-droid/16.0-r0/git $ ./git-version-gen .tarball-version
  14.2.88-52-g80015ed

* relevant parts from configure.ac:
  AC_INIT([pulseaudio-modules-droid], [m4_esyscmd(./git-version-gen .tarball-version)], [mer-general@lists.merproject.org])
  m4_define(pa_module_version, `echo $VERSION | cut -d. -f3 | cut -d- -f1`)

  so for this revision:
  PACKAGE_VERSION: 14.2.88-52-g80015ed
  PA_MODULE_VERSION: 88 (used only by common/libdroid-util.pc.in:Version: @PA_MODULE_VERSION@)

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
Upstream-Status: Pending
---
 src/common/Makefile.am          | 2 +-
 src/droid/Makefile.am           | 2 +-
 src/droid/module-droid-card.c   | 2 +-
 src/droid/module-droid-sink.c   | 2 +-
 src/droid/module-droid-source.c | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/common/Makefile.am b/src/common/Makefile.am
index fe5df8d..de2d87b 100644
--- a/src/common/Makefile.am
+++ b/src/common/Makefile.am
@@ -8,7 +8,7 @@ AM_CFLAGS = \
 	$(DROIDHEADERS_CFLAGS) \
 	$(HYBRIS_CFLAGS) \
 	$(EXPAT_CFLAGS) \
-	-DPULSEAUDIO_VERSION=@PA_MAJOR@ \
+	-DPULSEAUDIO_VERSION=\"@PA_MAJORMINOR@\" \
 	-I$(top_srcdir)/src/common \
 	-I$(top_srcdir)/src/common/include
 
diff --git a/src/droid/Makefile.am b/src/droid/Makefile.am
index fa117f5..f819852 100644
--- a/src/droid/Makefile.am
+++ b/src/droid/Makefile.am
@@ -7,7 +7,7 @@ AM_CFLAGS = \
 	$(PULSEAUDIO_CFLAGS) \
 	$(DROIDHEADERS_CFLAGS) \
 	$(HYBRIS_CFLAGS) \
-	-DPULSEAUDIO_VERSION=@PA_MAJOR@ \
+	-DPULSEAUDIO_VERSION=\"@PA_MAJORMINOR@\" \
 	-I$(top_srcdir)/src/droid \
 	-I$(top_srcdir)/src/common/include
 
diff --git a/src/droid/module-droid-card.c b/src/droid/module-droid-card.c
index 68baa6d..80a699c 100644
--- a/src/droid/module-droid-card.c
+++ b/src/droid/module-droid-card.c
@@ -66,7 +66,7 @@
 
 PA_MODULE_AUTHOR("Juho Hämäläinen");
 PA_MODULE_DESCRIPTION("Droid card jb2q");
-PA_MODULE_VERSION(PACKAGE_VERSION);
+PA_MODULE_VERSION(PULSEAUDIO_VERSION);
 PA_MODULE_USAGE(
         "card_name=<name for the card> "
         "sink_name=<name for the sink> "
diff --git a/src/droid/module-droid-sink.c b/src/droid/module-droid-sink.c
index 13546cd..48e178f 100644
--- a/src/droid/module-droid-sink.c
+++ b/src/droid/module-droid-sink.c
@@ -45,7 +45,7 @@ PA_MODULE_AUTHOR("Juho Hämäläinen");
 PA_MODULE_DESCRIPTION("Droid sink jb2q");
 PA_MODULE_USAGE("master_sink=<sink to connect to> "
                 "sink_name=<name of created sink>");
-PA_MODULE_VERSION(PACKAGE_VERSION);
+PA_MODULE_VERSION(PULSEAUDIO_VERSION);
 
 static const char* const valid_modargs[] = {
     "config",
diff --git a/src/droid/module-droid-source.c b/src/droid/module-droid-source.c
index ffa2cbf..ddb3bfd 100644
--- a/src/droid/module-droid-source.c
+++ b/src/droid/module-droid-source.c
@@ -44,7 +44,7 @@ PA_MODULE_AUTHOR("Juho Hämäläinen");
 PA_MODULE_DESCRIPTION("Droid source jb2q");
 PA_MODULE_USAGE("master_source=<source to connect to> "
                 "source_name=<name of created source>");
-PA_MODULE_VERSION(PACKAGE_VERSION);
+PA_MODULE_VERSION(PULSEAUDIO_VERSION);
 
 static const char* const valid_modargs[] = {
     "config",
